app/models/nic/base.rb

```diff
@@ -64,6 +64,13 @@ class Base < ApplicationRecord
 
     belongs_to_host :inverse_of => :interfaces, :class_name => "Host::Base"
 
+    scoped_search :on => :mac, :complete_value => true
+    scoped_search :on => :ip, :complete_value => true
+    scoped_search :on => :name, :complete_value => true
+    scoped_search :on => :managed, :complete_value => {:true => true, :false => false}
+    scoped_search :on => :primary, :complete_value => {:true => true, :false => false}
+    scoped_search :on => :domain, :complete_value => true
```

Comment:
These should all include `:only_explicit => true` option, otherwise searches without specifying the attribute to search on may cause very slow queries. 
Also, since this is only searchable from the API, there is no need for `:complete_value` option, though it does not hurt.

Reply:
This comment wasn't addressed